### ISSUE TITLE
Python: New Feature: Is it possible to run native functions while streaming?

### ISSUE THREAD

**Issue by tyler-suard-parker** [microsoft/semantic-kernel#6831]
I am seeing notebooks for calling functions and plugins, which is great (https://github.com/microsoft/semantic-kernel/blob/main/python/samples/getting_started/11-streaming-completions.ipynb), and some notebooks for streaming (https://github.com/microsoft/semantic-kernel/blob/main/python/samples/getting_started/08-native-function-inline.ipynb), which is great too.  Is there a way to call native functions while streaming?

**Comment by eavanvalkenburg** [comment]
What do you mean with native functions while streaming? Currently if you are using function calling and streaming, the package gathers all the function call responses to make the whole function call object, it then executes that function, regardless of what type of function it is, so that includes native functions, (it does those non-streaming) and then it calls the model again, with the results of the executed functions and that is streamed to the end-user again. So with streaming enabled and auto_invoke function calling enabled, you will see the function call stream coming, then it pauses while the functions are executed and then it continues streaming with the next response. Does that answer your question? @tyler-suard-parker

**Comment by tyler-suard-parker** [comment]
Yes, that does @eavanvalkenburg .  Thank you.  I did not know if function calling while streaming was possible in SK, and I didn't see any examples for that, so your answer is very helpful to me.

**Comment by tyler-suard-parker** [comment]
@eavanvalkenburg can SK call multiple functions asynchronously?

**Comment by eavanvalkenburg** [comment]
@tyler-suard-parker what do you mean exactly? two possible ways of answering your question: 1) the function calls might return multiple ones, those are all gathered and then executed in parallel, the model decided if parallel works (it will only supply that when the answer of one isn't needed as input for another, if that is the case it will supply the second in a second round) 2) currently kernel.invoke calls a single function, you could parallelize that by just doing a asyncio.gather in your own code, but if you do that with some shared variables then I have no clue how that will work, never tried it!

**Comment by tyler-suard-parker** [comment]
@eavanvalkenburg what I mean is, the user enters a question, "Should I invest in NVIDIA or Tesla?"

Semantic Kernel has a function defined called "get_stock_price_history(stock_symbol)" that can get the price history for one stock.

Semantic Kernel decides to call this function twice, 
get_stock_price_history("TSLA")
and
get_stock_price_hitsory("NVDA")

and the two functions run at the same time, rather than one after the other.  Is that possible with SK?

**Comment by nmoeller** [comment]
Hey Tyler,

i think you opened a new issue for this here, die anybody who wants to knew more about this please check out this issue here #6896
